### PR TITLE
 Fix error in session.js when an /app request fails 

### PR DESCRIPTION
**Pull request by robertknight**
When a request to the session endpoint fails with a network
error, avoid trying to parse the response in process().

This is the most frequently reported Sentry event - https://app.getsentry.com/hypothesis/prod-client/issues/116030860/

### CODE REVIEW

File: h/static/scripts/app.js [hunk 1]
```diff
@@ -28,30 +28,33 @@ if (settings.raven) {
...
-        threading.root = mail.messageContainer();
-
-        // Reload all new, unsaved annotations
-        threading.thread(drafts.unsaved());
-
-        return threading;
-      }
-    ]
-  };
+// Fetch external state that the app needs before it can run. This includes the
```
Comment (robertknight) [inline h/static/scripts/app.js]: This is just a cleanup of this function, no code changes other than using `@ngInject`. See the first commit in the PR.

File: h/static/scripts/session.js [hunk 2]
```diff
@@ -154,8 +154,11 @@ function session($http, $resource, $rootScope, flash, raven, settings) {
     return model;
   };
 
-  function process(data) {
-    // Parse as json
+  function process(data, headersGetter, status) {
+    if (status < 200) {
```
Comment (nickstenning) [inline h/static/scripts/session.js]: What about if `status == 504`?

Comment (robertknight) [inline h/static/scripts/session.js]: Good point! - Presumably any status >= 500 may also not be valid JSON?

**Comment by chdorner** [comment]
LGTM.

**Pull request merged**